lib/tty/pager/system.rb

```
@@ -123,22 +124,57 @@ def initialize(**options)
       #
       # @api public
       def page(text, &_callback)
-        return text unless output.tty?
+        write(text)
+        wait
+      end
+
+      # Spawn the pager process
+      #
+      # @return [PagerIO]
+      #   A wrapper for the external pager
+      #
+      # @api private
+      def start
+        # In case there's a previous pager running:
+        wait
 
         command = pager_command
         out = self.class.run_command(command)
         # Issue running command, e.g. unsupported flag, fallback to just command
-        if !out.empty?
+        unless out.empty?
           command = pager_command.split.first
         end
 
-        pager_io = open("|#{command}", 'w')
-        pid      = pager_io.pid
+        PagerIO.new(command)
+      end
 
-        pager_io.write(text)
-        pager_io.close
+      # Send text to the pager process. Starts a new process if it hasn't been
+      # started yet. Returns false if the pager was closed.
+      #
+      # @return [Boolean]
+      #   the success status of writing to the pager process
+      #
+      # @api public
+      def write(text)
+        @pager_io ||= start
+        @pager_io.write(text)
+        true
+      rescue Errno::EPIPE
```

I'd hide the error inside the `@pager_io`, what do you think?